### PR TITLE
fix: improve changelog parser

### PR DESCRIPTION
**Pull request by stefancplace**

### CODE REVIEW

**Comment by stefancplace** [comment]
detailed information can be found [here](https://collaborationfactory.slack.com/archives/C1GJZFF60/p1637770173204500)

**Review by csemrau — APPROVED** [review]

**Review by csemrau — APPROVED** [review]
Code change looks good. Did not test.

File: src/commands/release-notes/ReleaseNotesMessagesFile.ts [hunk 1]
```diff
@@ -19,16 +19,16 @@ export class ReleaseNotesMessagesFile {
     public static readonly DIRECTORY_RELEASE_NOTES: string = 'release-notes';
 
     private static readonly MESSAGES_FILE_NAME_PATTERN: RegExp = new RegExp(/^messages_.*\.db$/);
-    private static readonly CHANGELOG_DEFAULT_MESSAGE_PATTERN: RegExp = new RegExp(/(^|\n)changelog:[^\S\n]*(([^\n]|\n(?!\n))*)/);
-    private static readonly RELEVANCE_PATTERNS: string[] = [
-        'merge pull request #\\d+', // GitHub Pull Request
-        '\\bchangelog\\b' // Explicit changelog marker
+    private static readonly CHANGELOG_DEFAULT_MESSAGE_PATTERN: RegExp = new RegExp(/(^|\n{2})\s*changelog:[^\S\n]*(([^\n]|\n(?!\n))*)/, 'i');
```
Comment (stophi-dev) [inline src/commands/release-notes/ReleaseNotesMessagesFile.ts:22]: Not sure if we should require two line breaks before the changelog. If the line starts with `changelog:`, this should normally be enough. For example, what about this commit message:

```
Short description of commit

Additional description of commit
changelog: [ISSUE-xyz] my change log
```
So this should not be used for changelog generation? So we might miss some intended changelog entries. And in doubt, it is safer to add the entry.
Maybe this is more a question to @csemrau, as he formulated the requirement.

But I am fine to leave it as it is now.

Comment (stophi-dev) [reply src/commands/release-notes/ReleaseNotesMessagesFile.ts:22]: Ah wait, if it is at the start of the line, it is always considered to be a changelog. Ok, so then it is all fine.

**Review by stophi-dev — APPROVED** [review]

File: src/commands/release-notes/ReleaseNotesMessagesFile.ts (hunk 1 above)
Comment (csemrau) [reply src/commands/release-notes/ReleaseNotesMessagesFile.ts:22]: Your example forms a 2-line paragraph, and some UIs may display it as a single line (don't know if it is actually done). So yes, it is considered to be in the middle of a paragraph and should not be considered.

My original intention long time ago was to start a paragraph with the marker and even be able to have a multi-line paragraph, where line breaks are just discarded in the resulting changelog entry.

The most common case I have seen so far is that the commit message is a single line starting with the marker, so there's no ambiguity there.

**Pull request merged**